src/stores/ExploreDataStore.ts

```diff
@@ -0,0 +1,130 @@
+import { makeAutoObservable } from "mobx";
+import { ConstraintInstance, TypeFormatTuple, WorkflowConfig, WorkflowSolution } from "./WorkflowTypes";
+import { makePersistable } from "mobx-persist-store";
+
+const emptyWorkflowConfig = () => {
+  return {
+    domain: undefined,
+    inputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple ],
+    outputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple],
+    constraints: [ {constraint: {id:"",label:""}} as ConstraintInstance ],
+    minSteps: 3,
+    maxSteps: 4,
+    timeout: 120,
+    solutionCount: 10,
+    run_id: ""
+  }
+}
+
+export class ExploreDataStore {
+
+  workflowConfig: WorkflowConfig = emptyWorkflowConfig();
+  workflowSolutions: WorkflowSolution[] = [];
+  isGenerating: boolean = false;
+  generationError: string = "";
+
+  constructor() {
+    makeAutoObservable(this, {}, { deep: true });
+    makePersistable(this, { 
+      name: "ExploreDataStore",
+      properties: ["workflowConfig", "workflowSolutions"],
+      storage: window.localStorage
+    });
+  }
+
+  inputsOutputsToJSON(values: TypeFormatTuple[], dataRoot: string, formatRoot: string) {
+    return values.filter(value => value[0] !== undefined && value[1] !== undefined && value[0]!.id !== "" && value[1]!.id !== "")
+      .map((value) => {
+        return {
+          [dataRoot]: [value[0]!.id.replace("http://edamontology.org/", "")],
+          [formatRoot]: [value[1]!.id.replace("http://edamontology.org/", "")]
+        };
+    });
+  }
+
+  configToJSON(config: WorkflowConfig): any {
+    const dataRoot: string = "data_0006";
+    const formatRoot: string = "format_1915";
+    const toolsRoot: string = "operation_0004";
```

Comment:
The same as before, while this works for this case study, it would be safer to rely on the roots specified in the configuration. We could discuss these design choices at some point.